server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregationBuilder.java

```diff
@@ -220,6 +239,14 @@ protected AggregatorFactory doBuild(
                     + "]"
             );
         }
+        if (context.isInSortOrderExecutionRequired()) {
```

Comment:
Maybe `context.isInSortOrderExecutionRequired()` check is just sufficient here?
This is what triggers the usage of `TimeSeriesIndexSearcher` and this index searcher is incompatible with `composite` aggregation. We can then just return `[composite] aggregation is incompatible with time series execution mode` as error or similar?

Reply:
I didn't know that `this index searcher is incompatible with composite aggregation`. Does that mean that composite aggregations cannot be executed on a time series index?

Reply:
I did it that way to avoid doing the validation if the index is not a time series index.

Reply:
If `context.isInSortOrderExecutionRequired()` returns `true` then that means the `TimeSeriesIndexSearcher` is used and composite aggregation doesn't work well this. Just doing this check should be enough to throw a validation error here.